### PR TITLE
MET-6139 Move rdf convertion classes from metis-framework.

### PR DESCRIPTION
**Pull request by stzanakis**
@jeortizquan
This is a first step of the code migration.
In this first iteration Tests(if any) are not migrated, packaging organization not optimized, reported issues not resolved.
With the first review, we just agree on the direction and then we continue the implementation on this PR.

@jochen-vermeulen In the end the todo I mentioned was actually from you in the end. So perhaps you have a say in this one as well?

### CODE REVIEW

**Comment by stzanakis** [comment]
@jeortizquan 
This is a first step of the code migration. 
In this first iteration Tests(if any) are not migrated, packaging organization not optimized, reported issues not resolved.
With the first review, we just agree on the direction and then we continue the implementation on this PR.

File: src/main/java/eu/europeana/metis/schema/convert/RdfDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+      EDM_HAS_VIEW + XPATH_IS_OEMBED_RESOURCE_CONDITION;
+
+  private final XPathExpressionWrapper getObjectExpression = new XPathExpressionWrapper(xPath -> xPath.compile(EDM_OBJECT));
+  private final XPathExpressionWrapper getHasViewExpression = new XPathExpressionWrapper(xPath -> xPath.compile(EDM_HAS_VIEW));
+  private final XPathExpressionWrapper getIsShownAtExpression = new XPathExpressionWrapper(
+      xPath -> xPath.compile(EDM_IS_SHOWN_AT));
+  private final XPathExpressionWrapper getIsShownByExpression = new XPathExpressionWrapper(
+      xPath -> xPath.compile(EDM_IS_SHOWN_BY));
+  private final XPathExpressionWrapper getOEmbedExpression = new XPathExpressionWrapper(
+      xPath -> xPath.compile(OEMBED_XPATH_CONDITION_HAS_VIEW + " | " + OEMBED_XPATH_CONDITION_IS_SHOWN_BY));
```
Comment (jochen-vermeulen) [inline src/main/java/eu/europeana/metis/schema/convert/RdfDeserializer.java:65]: Media Processing specific code

File: src/main/java/eu/europeana/metis/schema/convert/RdfDeserializer.java [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+                    .collect(Collectors.toSet());
+  }
+
+  private Set<String> getOEmbedUrls(Document document) throws RdfDeserializationException {
+    final NodeList oEmbedNodes = getOEmbedExpression.evaluate(document);
+    return IntStream.range(0, oEmbedNodes.getLength())
+                    .mapToObj(oEmbedNodes::item)
+                    .map(Node::getNodeValue)
+                    .collect(Collectors.toSet());
+  }
```
Comment (jochen-vermeulen) [inline src/main/java/eu/europeana/metis/schema/convert/RdfDeserializer.java:229]: Media Processing specific code.

**Comment by jochen-vermeulen** [comment]
@stzanakis Just wondering about separation of concerns and all that. I think when it comes to the metis-schema module, we should try to keep it focused only on functionality that is specific to the RDF schema and converting to/from it.

It seems that you're proposing to move media-processing specific code to the metis schema (e.g. extract the hasView, isShownAt, etc.  from records). Reading back my TODO I believe my thoughts at the time was to transfer just the code that deserializes to a `Document`. The code specific to media processing could stay.

Specifically, in `RdfDeserializer` I marked the code that I think should stay in the media processing module. I think that `RdfNamespaceContext`, `ResourceEntry`, `RdfResourceEntry`, `RdfXpathContstants`, `UrlType` and perhaps other bits of code then also should stay in media processing?

Bits and pieces like `XPathExpressionWrapper` may perhaps stay and could then be used throughout the codebase? Or perhaps it should move to metis common instead? 

What do you think?

File: src/main/java/eu/europeana/metis/schema/convert/RdfDeserializer.java [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+   */
+  public RDF deserialize(InputStream inputStream) throws RdfDeserializationException {
+    try {
+      final IUnmarshallingContext context = rdfBindingFactory.createUnmarshallingContext();
+      return (RDF) context.unmarshalDocument(inputStream, UTF8);
+    } catch (JiBXException e) {
+      throw new RdfDeserializationException(
+          "Something went wrong with converting to or from the RDF format.", e);
+    }
+  }
```
Comment (jochen-vermeulen) [inline src/main/java/eu/europeana/metis/schema/convert/RdfDeserializer.java:122]: Duplicate with `RdfConversionUtils`

File: src/main/java/eu/europeana/metis/schema/convert/RdfSerializer.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+      IMarshallingContext context = rdfBindingFactory.createMarshallingContext();
+      context.setIndent(INDENTATION_SPACE);
+      ByteArrayOutputStream out = new ByteArrayOutputStream();
+      context.marshalDocument(rdf, UTF8, null, out);
+      return out.toByteArray();
+    } catch (JiBXException e) {
+      throw new RdfSerializationException(
+          "Something went wrong with converting to or from the RDF format.", e);
+    }
+  }
```
Comment (jochen-vermeulen) [inline src/main/java/eu/europeana/metis/schema/convert/RdfSerializer.java:86]: Duplicate with `RdfConversionUtils`.

**Comment by jochen-vermeulen** [comment]
And I think my TODO suggested something else. Namely, removing duplicate code between `RdfConversionUtils` already existing in Metis Schema on the one hand, and `RdfDeserializer` and `RdfSerializer` on the other. I marked two such cases, but I think there are many more opportunities for merging these three classes.

**Comment by stzanakis** [comment]
> @stzanakis Just wondering about separation of concerns and all that. I think when it comes to the metis-schema module, we should try to keep it focused only on functionality that is specific to the RDF schema and converting to/from it.
> 
> It seems that you're proposing to move media-processing specific code to the metis schema (e.g. extract the hasView, isShownAt, etc. from records). Reading back my TODO I believe my thoughts at the time was to transfer just the code that deserializes to a `Document`. The code specific to media processing could stay.
> 
> Specifically, in `RdfDeserializer` I marked the code that I think should stay in the media processing module. I think that `RdfNamespaceContext`, `ResourceEntry`, `RdfResourceEntry`, `RdfXpathContstants`, `UrlType` and perhaps other bits of code then also should stay in media processing?
> 
> Bits and pieces like `XPathExpressionWrapper` may perhaps stay and could then be used throughout the codebase? Or perhaps it should move to metis common instead?
> 
> What do you think?

Okay. As I mentioned earlier, the TODO comment was clear I just thought taking it a step further and move more code that could be potentially used by other projects. That being said I was a bit hesitant of this impactful change.
So that clears it I guess. 
Thanks for having a look.
Closing this to rework on a smaller update and perhaps some renaming on the media library.

**Pull request closed without merging**